hazelcast/src/main/java/com/hazelcast/instance/impl/NodeExtension.java

```diff
@@ -405,4 +406,6 @@ default boolean isClientFailoverSupported() {
     /** Returns the internal jet service backend */
     @Nullable
     JetServiceBackend getJetServiceBackend();
+
+    Object createSslEngineFactory(SSLConfig sslConfig);
```

Comment:
`SSLEngineFactory` is defined in the EE codebase, so the type is not available here.

Reply:
Could you add a line of comment to prevent future confusion?

@kwart or should we just move the SSLENgineFactory interface to OS?

Reply:
I would rather move the interface to OS.

Reply:
Get this verified with engineering management because it could be that they are not to keen on doing this.

Reply:
I think we should let it remain an object. I already tried to get some of this code moved to OS and I ran into a lot of resistance. So perhaps not waste time on this any further an accept this ugly API.

Reply:
This change (i.e. moving one interface) to OS side seems safe to me. I don't think it could break anything. In the EE JAR it should stay on the same place as it is now. 
So I would rather try to keep this close in both branchech 5.3.z and master

Reply:
Added another commit that moves the interface to OS, as requested